From 052232ac12890152fc3873d94a91f7279317b1cf Mon Sep 17 00:00:00 2001
From: Andrey Dolnikov <andrey.dolnikov@cogentembedded.com>
Date: Wed, 7 Oct 2020 18:13:39 +0300
Subject: [PATCH 1010/1013] gpu: drm: rcar-du: rcar_du_vsp: Check if gem buffer
 has a physical address

Do not copy the sg in case gem buffer does have a physical address.

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_vsp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_vsp.c b/drivers/gpu/drm/rcar-du/rcar_du_vsp.c
index b9317ad..0db759d 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_vsp.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_vsp.c
@@ -197,7 +197,7 @@ int rcar_du_vsp_map_fb(struct rcar_du_vsp *vsp, struct drm_framebuffer *fb,
 		struct drm_gem_cma_object *gem = drm_fb_cma_get_gem_obj(fb, i);
 		struct sg_table *sgt = &sg_tables[i];
 
-		if (gem->sgt) {
+		if (gem->sgt && !gem->paddr) {
 			struct scatterlist *src;
 			struct scatterlist *dst;
 
-- 
2.7.4

